src/logger/POSTLogger.ts

```diff
@@ -2,35 +2,44 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import type { Destroyable } from '../destroyable/Destroyable';
-import LogLevel from '../logger/LogLevel';
-import MeetingSessionConfiguration from '../meetingsession/MeetingSessionConfiguration';
 import IntervalScheduler from '../scheduler/IntervalScheduler';
 import Log from './Log';
+import Logger from './Logger';
+import LogLevel from './LogLevel';
 
 /**
- * `MeetingSessionPOSTLogger` publishes log messages in batches to a URL
+ * `POSTLogger` publishes log messages in batches to a URL
  * supplied during its construction.
  *
- * Be sure to call {@link MeetingSessionPOSTLogger.destroy} when you're done
+ * Be sure to call {@link POSTLogger.destroy} when you're done
  * with the logger in order to avoid leaks.
  */
-export default class MeetingSessionPOSTLogger implements Destroyable {
+export default class POSTLogger implements Logger, Destroyable {
   private logCapture: Log[] = [];
   private sequenceNumber: number = 0;
   private lock = false;
   private intervalScheduler: IntervalScheduler;
   private eventListener: undefined | (() => void);
+  private headers: Record<string, string>;
+  metadata: Record<string, string>;
 
   constructor(
-    private name: string,
-    private configuration: MeetingSessionConfiguration,
     private batchSize: number,
     private intervalMs: number,
     private url: string,
     private level = LogLevel.WARN,
-    private headers?: Record<string, string>
+    options?: {
```

Comment:
For consistency, we have two options.

1. `batchSize`, `intervalMs`, `url`, and `level` can also be under `options`. (Especially, `level` is also an optional field.)

    For example, we had to specify `LOGGER_BATCH_SIZE` and `LOGGER_INTERVAL_MS` to indicate 1st and 2nd parameters. With options, we won't need these comments.

    ```js
    const logger = new POSTLogger(
      20, // LOGGER_BATCH_SIZE
      2000, // LOGGER_INTERVAL_MS
      URL,
      LogLevel.INFO,
      postLoggerOptions
    );
    ```
2. Or you can make `metadata` and `headers` as individual parameters. JS SDK uses this pattern more.

Reply:
Used the first approach since we have many properties some optional and some with default value and only URL as required. Thanks for the suggestion!